### PR TITLE
Feature/quickstart osx

### PR DESCRIPTION
**Pull request by prashantmital**
QuickStart supported for OS X users (who have conda + port/brew installed) with the `quickstart` executable script.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -18,7 +18,7 @@ before_install:
     - conda config --set always_yes yes --set changeps1 no
     - conda update -q conda
     - conda info -a
-    - conda create -q -n test-environment python=$PYVERSION setuptools pip numpy=1.8.2 pyzmq ipython matplotlib cython sphinx mock coverage requests pyyaml mpi4py
+    - conda create -q -n test-environment python=$PYVERSION setuptools pip numpy=1.9.2 pyzmq ipython matplotlib cython sphinx mock coverage requests pyyaml mpi4py
```
Comment (prashantmital) [inline .travis.yml]: Not sure why this changed. Please ignore for the purposes of reviewing this PR.

File: utils/distarray_conda-env.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+name: distarray
+dependencies:
```
Comment (bgrant) [inline utils/distarray_conda-env.yml]: Can these be narrowed down to the top-level dependencies (and let conda figure out the rest)?

Comment (prashantmital) [inline utils/distarray_conda-env.yml]: This file is actually auto generated by conda, so I assume that can be done. I will run some tests and update this.

Comment (prashantmital) [inline utils/distarray_conda-env.yml]: Ok so it turns out the whole point of being able to specify environments with the `yml` file is that you can EXACTLY specify the environment. Conda does not resolve any dependencies when this route is taken. So when I removed all but the things I had installed at the command lines, conda didn't even install things like `setuptools`. 

However I see your point that if we stick to this methodology we will have to keep updating this `yml` file for the quick start to remain current. I will try to devise an alternative strategy.

File: quickstart [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+    resolve_conda
+    r_stat=$?
+    exit_install $r_stat
+else
+    echo "DistArray quickstart is only supported on OSX."
+    exit 1
+fi
+
+conda create -n $CONDA_ENV numpy ipython ipython-notebook cython sphinx mock
```
Comment (prashantmital) [inline quickstart]: So we no longer rely on a file.

**Comment by bgrant** [comment]
Hi Prashant- looks interesting.  I'm trying to get to a point where I can try it out.

Do you have docs to go along with this?  At minimum, a comment at the top of the file saying what it does would be good.  If this is for OS X and conda only, I would recommend renaming it to something like "quickstart-osx-conda.sh" or some such.

Is it Python 2 only, or will it work for Python 3 as well?  For expansion, do you see us adding more scripts or adding cases to this script?

Thanks.

**Comment by bgrant** [comment]
This isn't a trivial addition, but it would be great to test this (and DistArray) on TravisCI.  They have some facility for testing on OS X, but we currently only test on Linux.

http://docs.travis-ci.com/user/osx-ci-environment/

Let me know if you're interested in setting this up.

**Comment by prashantmital** [comment]
New branch for quick start. Closing PR #614

**Pull request closed without merging**